### DISCUSSION TITLE
How to use the coverage weight option inside coverpoint?

### DISCUSSION THREAD

**Discussion by aneels3** [fvutils/pyvsc#113] [Q&A]
Hi @mballance,
I am trying to implement the coverage weight option within a cover point.
I need to implement the below SV implementation using PyVSC coverage.
![image](https://user-images.githubusercontent.com/32580268/133620442-e38f5813-8a31-4028-b1a3-0ecbc1178514.png)
    
```python3
    @vsc.covergroup
    class fcvt_s_w_cg(object):
        def __init__(self, precision, sign_typ):
            super().__init__()

            self.instr = None
            self.cp_rs1 = vsc.coverpoint(lambda: self.instr.rs1,
                                         cp_t=vsc.enum_t(riscv_reg_t))
            self.cp_fd = vsc.coverpoint(lambda: self.instr.fd,
                                        cp_t=vsc.enum_t(riscv_fpr_t))
            self.cp_rs1_sign = vsc.coverpoint(lambda: self.instr.rs1_sign,
                                              cp_t=vsc.enum_t(operand_sign_e))
                                   # How can I specify weight option w.r.t sign_typ here?
            self.cp_fd_sign = vsc.coverpoint(lambda: self.instr.fd_sign,
                                             cp_t=vsc.enum_t(operand_sign_e))
            self.cp_gpr_hazard = vsc.coverpoint(lambda: self.instr.gpr_hazard,
                                                cp_t=vsc.enum_t(hazard_e))
                                                
```
I have tried adding there but was ended up with a syntax error.
Can you help me on this?

Thanks and Regards,
Anil

**Comment by mballance** [comment]
Hi @aneels3,
    Coverpoint options are configured by attaching a 'dict' to the coverpoint. The testcase below shows overriding the `at_least` option for one coverpoint:
https://github.com/fvutils/pyvsc/blob/612de9e6244c685a3df1972e4860abfe35b614e1/ve/unit/test_covergroup_options.py#L58-L75

I checked the documentation and realized that it was a bit out of date, so I've made a few updates there as well.

Best Regards,
Matthew
